### ISSUE TITLE
Visual editor incorrectly rewrites escaping of dashes

### ISSUE THREAD

**Issue by ailich** [rstudio/rstudio#10996]
Very similar to rstudio/rstudio#10012, but the same behavior occurs with dashes when switching between visual and source editor.

For example, if you enter `\-\-` in source editor this should render as `--` and it does. However, if at any point you switch to visual editor the escaping characters (`\`) are removed so it renders as a single long dash instead (`–`).

**Comment by ronblum** [comment]
@ailich Thank you for raising this! I suspect that this behavior is driven by Pandoc as with #10012, so I'll mark it for review.

**Comment by jjallaire** [comment]
Should be fixed here: https://github.com/rstudio/rstudio/commit/d61546bb4ef37a7272f9728d0053125f8f870420

**Comment by jjallaire** [comment]
That's expected (pandoc only escapes the first dash because that's all that's needed for disambiguation). You can see this with raw pandoc roundtripping as follows:

```bash
$ echo "\\-\\-" | pandoc --to json | pandoc --from json  --to markdown
\--
```

**Comment by romainfrancois** [comment]
Thanks. Then I can confirm this is fixed with RStudio 2022.06.0-daily+369. 

<img width="942" alt="image" src="https://user-images.githubusercontent.com/2625526/167243595-af48b738-77a6-4763-8494-6f778ab102f7.png">

<img width="925" alt="image" src="https://user-images.githubusercontent.com/2625526/167243604-51382862-f5b7-4d44-b801-493fbe3f4ca2.png">

<img width="949" alt="image" src="https://user-images.githubusercontent.com/2625526/167243618-210f0b2e-b177-4c52-a44a-c2de7a36add2.png">
